samples/archivist_story_runner.py

```diff
@@ -0,0 +1,305 @@
+"""archivist story runner takes a yaml config file listing operations to be conducted serially.
+
+It then runs the defined operations.
+"""
+
+import time
+from enum import Enum
+import argparse
+import yaml
+
+# remove warning from console
+import urllib3
+
+from archivist.compliance_polices import PolicyType
+from archivist.archivist import Archivist
+
+# stop printing to console that the request is insecure
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+class Operation(Enum):
+    """
+    Operation are the allowed archivist operations to be run.
+    """
+
+    CREATE_ASSET = 1
+    CREATE_EVENT = 2
+    CREATE_COMPLIANCE_POLICY = 3
+    CHECK_COMPLIANCE = 4
+    DELETE_COMPLIANCE = 5
+
+
+def parse_arg(arg, key, default=None):
+    """
+    parse_arg parses a dictionary arg, and returns the value of a key.
+              if the key doesn't exist within the dictionary the default is returned.
+              if the default is None, then KeyError is raised.
+    """
+    try:
+        return arg[key]
+    except KeyError as ke:
+        if default is None:
+            raise ke
+
+        return default
+
+
+class ArchivistStoryRunner():
+    """
+    ArchivistStoryRunner takes a url, token_file and a yaml config_file.
+
+    The yaml config_file contains a list of `operations` to be performed serially, e.g.
+
+    ```
+    operations:
+
+        - operation: CREATE_ASSET
+            to_print: Create an empty radiation bag with id 1.
+            wait_time: 10
+            asset_id: radiation bag 1
+            behaviours:
+              - RecordEvidence
+            attributes:
+              radioactive: "true"
+              radiation_level: "0"
+              weight: "0"
+    ```
+
+    where:
+     - `operation` is the operation to perform (see Operation Enum for options).
+     - `to_print` is what to print to the console.
+     - `wait_time` is time to wait before running the operation.
+     - `asset_id` is the local reference to the asset, so can be referenced in other operations.
+     - `behaviours` are a list of asset behaviours
+     - `attributes are the asset's attributes
+
+    please see example yaml files for other operation examples.
+
+    To perform all the operations call the `run` method.
+    """
+
+    assets = dict()  # dict of assets created
+    events = list()  # list of events created
+    policies = dict()  # dict of policies created
+
+    def __init__(self, url, token_file, config_file):
```

Comment:
not a good idea to pass filenames as arguments - leads to testing difficulties and restricts your user.
Rather accept a fd iterator as an argument and expect your ise to open the file and close it correctly.

See https://softwareengineering.stackexchange.com/questions/262346/should-i-pass-in-filenames-to-be-opened-or-open-files
for example